core/src/main/scala/codesearch/core/index/repository/Extensions.scala

```diff
@@ -0,0 +1,29 @@
+package codesearch.core.index.repository
+
+trait Extension[A] {
+  def extensions: Set[String]
+}
+
+object Extensions {
+
+  implicit def jsExtensions[A <: JavaScript]: Extension[A] = new Extension[A] {
+    override def extensions: Set[String] =
+      Set("js", "ts", "json", "xml", "yml", "coffee", "markdown", "md", "yaml", "txt")
```

Comment:
Hm, this is actually wrong. We need two sets of extensions – one to figure out which files to leave after downloading the package, and another set to determine which files are counted as "source files" as far as the relevant checkbox is concerned.

If my concern is valid, please create a separate issue for that.

Reply:
> please create a separate issue for that

I've done that (#93)